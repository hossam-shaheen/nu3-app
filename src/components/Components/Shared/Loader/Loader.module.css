.loader {
    color: var(--secondary);
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    font-size: 28px;
    margin-top: 15px;
    margin-bottom: 15px;
}