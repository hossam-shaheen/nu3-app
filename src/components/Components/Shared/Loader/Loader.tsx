import { FunctionComponent } from "react";
import classes from "./Loader.module.css";

export const Loader: FunctionComponent = (): JSX.Element => {
  return (
    <div className={classes["loader"]} data-testid="loader">
      Loading...
    </div>
  );
};
export default Loader;
