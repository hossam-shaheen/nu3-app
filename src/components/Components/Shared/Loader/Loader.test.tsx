import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

test('Loader  : Renders', () => {
    render(<Loader />);
    const loader = screen.queryByTestId('loader');
    expect(loader).toBeInTheDocument();
});



test('Loader: Snapshot', () => {
    const component = renderer.create(<Loader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});