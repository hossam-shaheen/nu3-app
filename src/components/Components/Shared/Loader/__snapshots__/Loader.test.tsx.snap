// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: Snapshot 1`] = `
<div
  className="loader"
  data-testid="loader"
>
  Loading...
</div>
`;
